### PR TITLE
comstogm Code Review 1 suggestions

### PR DESCRIPTION
**Pull request by comstogm**
Ideastorm is an application that takes in a few modifiers to a number of participants in a group, desired activity description, and general cost of said activity to return a suggested activity using the bored api. 

The program was available on GitHub in time. 

The program was well commented and easy to understand.

The program compiled.

I learned more clearly about practices revolving around annotations in Kotlin code. The coding conventions page suggests that annotations should be on separate lines before the declaration to which they are attached with the same indentation.
https://kotlinlang.org/docs/coding-conventions.html#annotations

The MainActivity of this application was composed mostly out of Spinners, which I have not yet interacted with much. These were very nicely implemented, and the formatting gave me an example to look at in the future. 

The Activity class was very nicely formatted, and looking at the Kotlin coding conventions, this is the way we should do this in my groups project. Classes with many properties should be located on different lines. https://kotlinlang.org/docs/coding-conventions.html#class-headers

### CODE REVIEW

File: app/src/main/java/com/ideastorm/v25001/IdeaStormApplication.kt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package com.ideastorm.v25001
 
 import android.app.Application
+import org.koin.android.BuildConfig
```
Comment (comstogm) [inline app/src/main/java/com/ideastorm/v25001/IdeaStormApplication.kt:4]: Imported BuildConfig. I was getting a redline under BuildConfig.DEBUG

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 2]
```diff
@@ -74,7 +74,7 @@ fun OptionMenu(appName: String) {
         backgroundColor = Color(android.graphics.Color.parseColor("#D9D9D9")),
         actions = {
             IconButton(onClick = { showMenu = !showMenu }) {
-                Icon(Icons.Default.MoreVert, "Navigation")
+                Icon(Icons.Default.MoreVert, stringResource(R.string.Navigation))
```
Comment (comstogm) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:77]: Extracted String resource as detailed in class videos to help future possible language support.

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 3]
```diff
@@ -115,7 +115,9 @@ fun Greeting(greeting: String) {
  */
 @Composable
 fun ParticipantsSpinner() {
-    val participantOptions = listOf("One person", "Two person", "Group")
+    val participantOptions = listOf(stringResource(R.string.OnePerson), stringResource(R.string.TwoPerson), stringResource(
+            R.string.Group)
+        )
```
Comment (comstogm) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:120]: Extracted String resource as detailed in class videos to help future possible language support.

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 4]
```diff
@@ -151,7 +153,11 @@ fun ParticipantsSpinner() {
  */
 @Composable
 fun ActivityTypeSpinner() {
-    val typeOptions = listOf("Education", "Recreational", "Social", "Diy", "Charity", "Cooking", "Relaxation", "Music", "Busywork")
+    val typeOptions = listOf(stringResource(R.string.Education), stringResource(R.string.Recreational), stringResource(
+            R.string.Social), stringResource(R.string.Diy), stringResource(R.string.Charity), stringResource(
+                    R.string.Cooking), stringResource(R.string.Relaxation), stringResource(R.string.Music), stringResource(
+                            R.string.Busywork)
+                        )
```
Comment (comstogm) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:160]: Extracted String resource as detailed in class videos to help future possible language support.

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 5]
```diff
@@ -187,7 +193,9 @@ fun ActivityTypeSpinner() {
  */
 @Composable
 fun PriceSpinner() {
-    val priceOptions = listOf("Free", "Low", "High")
+    val priceOptions = listOf(stringResource(R.string.Free), stringResource(R.string.Low), stringResource(
+            R.string.High)
+        )
```
Comment (comstogm) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:198]: Extracted String resource as detailed in class videos to help future possible language support.

File: app/src/main/java/com/ideastorm/v25001/RetrofitClientInstance.kt [hunk 6]
```diff
@@ -17,7 +17,6 @@ object RetrofitClientInstance {
                     .baseUrl(BASE_URL)
                     .addConverterFactory(GsonConverterFactory.create())
                     .build()
-                print("fo")
```
Comment (comstogm) [inline app/src/main/java/com/ideastorm/v25001/RetrofitClientInstance.kt:20]: removed extraneous print command from RetrofitClientInstance.

File: app/src/main/java/com/ideastorm/v25001/dto/Activity.kt [hunk 7]
```diff
@@ -2,13 +2,20 @@ package com.ideastorm.v25001.dto
...
+                    @SerializedName("type")
+                    var type: String,
+                    @SerializedName("participants")
+                    var participants: Int,
+                    @SerializedName("price")
+                    var price: Float,
+                    @SerializedName("link")
+                    var link: String,
+                    @SerializedName("key")
+                    var key: Long) {
```
Comment (comstogm) [inline app/src/main/java/com/ideastorm/v25001/dto/Activity.kt:18]: Moved annotation to a separate line as detailed in coding conventions:
https://kotlinlang.org/docs/coding-conventions.html#annotations

File: app/src/main/res/values/strings.xml [hunk 8]
```diff
@@ -3,4 +3,20 @@
...
+    <string name="Social">Social</string>
+    <string name="Diy">Diy</string>
+    <string name="Charity">Charity</string>
+    <string name="Cooking">Cooking</string>
+    <string name="Relaxation">Relaxation</string>
+    <string name="Music">Music</string>
+    <string name="Busywork">Busywork</string>
+    <string name="Free">Free</string>
+    <string name="Low">Low</string>
+    <string name="High">High</string>
```
Comment (comstogm) [inline app/src/main/res/values/strings.xml:21]: Extracted String resource as detailed in class videos to help future possible language support.

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/com/ideastorm/v25001/MainActivity.kt:77]: Good.

File: app/src/main/java/com/ideastorm/v25001/RetrofitClientInstance.kt (hunk 6 above)
Comment (discospiff) [reply app/src/main/java/com/ideastorm/v25001/RetrofitClientInstance.kt:20]: I'd replace this with a log statement.  It looks like that's what the group is trying to do here.

File: app/src/main/java/com/ideastorm/v25001/dto/Activity.kt (hunk 7 above)
Comment (discospiff) [reply app/src/main/java/com/ideastorm/v25001/dto/Activity.kt:18]: The IDE can format code automatically, and consistently, for all developers on the team.   No need to do it manually.  :)

**Comment by discospiff** [comment]
String extraction is a good idea.  I recommend merging this branch.

**Pull request merged**